### ISSUE TITLE
authorizedToCreate seems to use cached data

### ISSUE THREAD

**Issue by JellowGerard** [laravel/nova-issues#3695]
- Laravel Version: 8.62.0
- Nova Version: 3.29.0
- PHP Version: 8.0.13
- Database Driver & Version: Eloquent
- Operating System and Version: MacOS 12.1
- Browser type and version: Firefox 95.0.2
- Reproduction Repository: private

### Introduction:
I have a resource Companies, which has a hasOne relationship with Subscriptions. Subscriptions has its own page, but this page should only be used to view all subscriptions, not to create a new subscription. This should be done via the Companies resource.

At first, I tried to achieve this with policies, but put that idea aside, as [this issue](https://github.com/laravel/nova-issues/issues/3177) was blocking me. Instead, I am now trying to do by adding the following to the Subscriptions resource:

```
public static function authorizedToCreate(Request $request)
{
    return $request->is('*/companies/*') || $request->get('viaResource') === 'companies';
}
```

This should show the create button for the relation on the companies resource page, and allow creation if the create page was opened via the Companies resource page, while it should remove the create button from the Subscription resource page.

### Issue:
It work only sometimes. For some reason, when navigating to one of the pages, it remembers what the last page was on which the permission was checked. Only after refreshing the page do the correct permissions apply.

E.g.:
- Go to Subscriptions
- Note that the button is not there
- Go to Companies > Details > Subscription
- The button is not there
- Refresh the page
- The button is there
- Go to Subscriptions (overview page)
- The button is there
- Refresh
- The button is gone

My best guess is that there is some kind of caching that is looked at while rendering the button, before re-checking the permission.

Hope you understand the issue and can help me resolve it.

**Comment by crynobone** [comment]
Duplicate of #2890 and #1893

Unfortunately this will only be fixed in Laravel Nova 4 as described in #1893

**Comment by monaye** [comment]
@crynobone could at least share a workaround for the Nova3 customer who can't upgrade to Nova 4? the issue was initially opened in 2019...
